ang/civicase/CaseOverview.js

```diff
@@ -1,73 +1,102 @@
 (function (angular, $, _) {
   var module = angular.module('civicase');
-  // Todo: To write unit tests
+
   module.directive('civicaseCaseOverview', function () {
     return {
       restrict: 'EA',
       replace: true,
       templateUrl: '~/civicase/CaseOverview.html',
       controller: civicaseCaseOverviewController,
-      scope: {}
+      scope: {},
+      link: civicaseCaseOverviewLink
     };
 
-    function civicaseCaseOverviewController ($scope, crmApi) {
-      $scope.caseStatuses = CRM.civicase.caseStatuses;
-      $scope.caseTypes = CRM.civicase.caseTypes;
-      $scope.caseTypesLength = _.size(CRM.civicase.caseTypes);
-      $scope.summaryData = [];
-
+    /**
+     * Link function for civicaseCaseOverview
+     *
+     * @param {Object} $scope
+     * @param {jQuery} element
+     * @param {Object} attrs
+     */
+    function civicaseCaseOverviewLink ($scope, element, attrs) {
       (function init () {
-        // We hide the breakdown when there's only one case type
-        if ($scope.caseTypesLength < 2) {
-          $scope.showBreakdown = false;
-        }
-
-        loadStatsData();
+        $scope.$watch('showBreakdown', showBreakdownWatcher);
       }());
 
       /**
-       * Creates link to the filtered cases list
-       *
-       * @param {String} type
-       * @param {String} status
-       * @return {String} link to the filtered list of cases
+       * Watchers for showBreakdown variable
        */
-      $scope.caseListLink = function (type, status) {
-        var cf = {};
+      function showBreakdownWatcher () {
```

Comment:
please tell what the function _does_, not what it _is_ (same for the documentation)